File: src/visualization/web_dashboard/frontend/api_client.ts
"""
API client utility for frontend-backend communication.
Provides functions to fetch metrics, alerts, health, and logs from backend API.

References:
- API endpoints in dashboard backend
- PEP 8, type hints, and docstring standards
"""
export async function fetchMetricsLive(): Promise<any> {
  const response = await fetch('/metrics/live');
  return response.json();
}

export async function fetchMetricsAverage(): Promise<any> {
  const response = await fetch('/metrics/average');
  return response.json();
}

export async function fetchAlerts(): Promise<any> {
  const response = await fetch('/alerts');
  return response.json();
}

export async function fetchHealth(): Promise<any> {
  const response = await fetch('/health');
  return response.json();
}

export async function fetchLogs(): Promise<any> {
  const response = await fetch('/logs');
  return response.json();
}

File: src/visualization/web_dashboard/api_client.py
"""
Dashboard API client for frontend integration.
Fetches metrics, alerts, and config from backend API.

References:
- Backend/frontend communication
"""
import requests

API_BASE = "http://localhost:8000/api"

def get_metrics():
    resp = requests.get(f"{API_BASE}/metrics")
    return resp.json()

def get_alerts():
    resp = requests.get(f"{API_BASE}/alerts")
    return resp.json()

def get_config():
    resp = requests.get(f"{API_BASE}/config")
    return resp.json()
